aws/resource_aws_dx_lag.go

```diff
@@ -0,0 +1,162 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDxLag() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDxLagCreate,
+		Read:   resourceAwsDxLagRead,
+		Update: resourceAwsDxLagUpdate,
+		Delete: resourceAwsDxLagDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"connections_bandwidth": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDxLagBandWidth,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"number_of_connections": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"force_destroy": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+		},
+	}
+}
+
+func resourceAwsDxLagCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	input := &directconnect.CreateLagInput{
+		ConnectionsBandwidth: aws.String(d.Get("connections_bandwidth").(string)),
+		LagName:              aws.String(d.Get("name").(string)),
+		Location:             aws.String(d.Get("location").(string)),
+		NumberOfConnections:  aws.Int64(int64(d.Get("number_of_connections").(int))),
+	}
+	resp, err := conn.CreateLag(input)
+	if err != nil {
+		return err
+	}
+	d.SetId(*resp.LagId)
+	return resourceAwsDxLagRead(d, meta)
+}
+
+func resourceAwsDxLagRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+
+	lagId := d.Id()
+	input := &directconnect.DescribeLagsInput{
+		LagId: aws.String(lagId),
+	}
+	resp, err := conn.DescribeLags(input)
+	if err != nil {
+		return err
+	}
+	if len(resp.Lags) != 1 {
+		if len(resp.Lags) < 1 {
+			d.SetId("")
```

Comment:
I think in this case (when Lag was removed) we don't want to error out - at least it's a convention not to do that.

Also - which is rather a nitpick - we can move this condition above or below the other condition and avoid the nesting & indentation so it's easier to read.

Reply:
I understood the convention 🙇